frontend/stores/user.js

```diff
@@ -0,0 +1,207 @@
+import { defineStore } from 'pinia'
+import { useGlobalStore } from './global'
+
+export const useUserStore = defineStore('user', {
+    state: () => {
+        return {
+            userData: {
+                token: null,
+                isAuthenticated: false,
+                pending: false,
+                error: '',
+                user: {
+                    id: null,
+                    email: '',
+                    username: ''
+                },
+            },
+            register: {
+                errorMessage: '',
+                succes: '',
+            },
+        }
+    },
+    getters: {
+        isAuthenticated: (state) => state.userData.isAuthenticated,
+    },
+    actions: {
+        /**
+         * See if user is logged-in
+         * Remember this only works client side, make user to use `if (process.client) {}` or the `onMounted()` hook in a vue component
+         */
+        async loadUser() {
+            this.userData.error = ''
+            this.userData.pending = true
+            const token = this.userData.token || localStorage.getItem('token')
+
+            const config = {
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+            }
+
+            if (token) {
+                config.headers['Authorization'] = `Token ${token}`
+            }
+
+            await $cmsApi('/api/auth/me/', config)
+                .then(response => {
+                    this.userData = {
+                        ...this.userData,
+                        isAuthenticated: true,
+                        user: response,
+                    }
+                })
+                .catch(error => {
+                    console.log('error //> ', error)
+                    this.resetUser()
+                    this.userData.error = error
+                    this.userData.isAuthenticated = false
+                }).finally(() => {
+                    this.userData.pending = false
+                })
+        },
+        /**
+         * Register user
+         * @param {username, email, password} body 
+         */
+        async registerUser(body) {
+            const global = useGlobalStore()
+
+            const config = {
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                method: 'POST',
+                body
+            }
+
+            const { data: register, pending, error } = await useAsyncData('register', () => $cmsApi('/api/auth/register/', config))
+
+            if (error.value) {
+                let warnMessage = null
+                for (const [key, value] of Object.entries(error._value.data)) {
+                    warnMessage = warnMessage ? `${warnMessage} \n\n ${key}: ${value}` : `${key}: ${value}`
+                }
+                this.register.succes = false
+                // Notification
+                global.negativeNotify(warnMessage)
```

Comment:
This error message can maybe in the future be clearer to show why registration failed (password too short, etc.)